pkg/localmetrics/localmetrics.go

```diff
@@ -269,3 +286,75 @@ func (c *MetricsCollector) AddAccountReuseCleanupFailure() {
 func (c *MetricsCollector) SetReconcileDuration(controller string, duration float64) {
 	c.reconcileDuration.WithLabelValues(controller).Observe(duration)
 }
+
+// AddAPICall observes metrics for a call to an external API
+// - param controller: The name of the controller making the API call
+// - param req: The HTTP Request structure
+// - param resp: The HTTP Response structure
+// - param duration: The number of seconds the call took.
+func (c *MetricsCollector) AddAPICall(controller string, req *http.Request, resp *http.Response, duration float64) {
+	c.apiCallDuration.With(prometheus.Labels{
+		"controller": controller,
+		"method":     req.Method,
+		"resource":   resourceFrom(req.URL),
+		"status":     resp.Status,
+	}).Observe(duration)
+}
+
+// resourceFrom normalizes an API request URL, including removing individual namespace and
+// resource names, to yield a string of the form:
+//     $group/$version/$kind[/{NAME}[/...]]
+// or
+//     $group/$version/namespaces/{NAMESPACE}/$kind[/{NAME}[/...]]
+// ...where $foo is variable, {FOO} is actually {FOO}, and [foo] is optional.
+// This is so we can use it as a dimension for the apiCallCount metric, without ending up
+// with separate labels for each {namespace x name}.
+func resourceFrom(url *neturl.URL) (resource string) {
+	defer func() {
```

Comment:
Can we avoid the `panic()/recover()` control flow? It might end up swallowing errors from other parts of the code later in life.

Reply:
Since it's limited to this function, it won't swallow anything higher in the call stack. And we very much want to swallow anything unexpected that happens here and below. If we don't, an unexpected error (think malformed URL resulting in [array index out of bounds](https://github.com/openshift/aws-account-operator/pull/410/files/c06c3833e04cc709adb3bc2f8d03260abeebda18#diff-6454382f68b967b06cd3ef5504f90bc1R77-R79)) would percolate up and `panic` the caller, which could be very bad.

Do you agree?

Reply:
Maybe we would want there to be a panic in some cases. Assume in the future if there is a new path `/api-x/openshift/` all the objects under it would get counted under the same metric and there would be no way to find out when it happens. But in the end it's a stylistic choice for me(I don't like jumps in execution). So feel free to ignore it.

Reply:
The catch-all will result in those calls being counted in the `{OTHER}` bucket. As noted in the comment, I'm trying not to solve a problem we don't know we have yet -- if we start seeing things in that bucket, we can start digging in.

Reply:
That works as well